en/Building_a_Simple_Engine/Loading_Models/08_animations.adoc

```diff
@@ -0,0 +1,1061 @@
+:pp: {plus}{plus}
+
+= Loading Models: Updating Animations
+
+== Understanding and Implementing Animations
+
+=== Introduction to 3D Animations
+
+Animation is a crucial aspect of modern 3D applications, bringing static models to life with movement and interactivity. In our engine, we've implemented a robust animation system that supports skeletal animations from glTF files.
+
+Animations in 3D graphics typically involve:
+
+* *Keyframes*: Specific points in time where the state of an object is explicitly defined
+* *Interpolation*: The process of calculating intermediate states between keyframes
+* *Channels*: Different properties that can be animated (position, rotation, scale)
+* *Bones/Joints*: A hierarchical structure that defines how parts of a model move together
+
+glTF provides a standardized way to store and transfer animations, which our engine can load and play back.
+
+=== Animation Data Structures
+
+As we saw in the link:03_model_system.adoc[Model System chapter], our engine uses several structures to represent animations:
+
+[source,cpp]
+----
+// Structure for animation keyframes
+struct AnimationChannel {
+    enum PathType { TRANSLATION, ROTATION, SCALE };
+    PathType path;
+    Node* node = nullptr;
+    uint32_t samplerIndex;
+};
+
+// Structure for animation interpolation
+struct AnimationSampler {
+    enum InterpolationType { LINEAR, STEP, CUBICSPLINE };
+    InterpolationType interpolation;
+    std::vector<float> inputs;  // Key frame timestamps
+    std::vector<glm::vec4> outputsVec4;  // Key frame values (for rotations)
+    std::vector<glm::vec3> outputsVec3;  // Key frame values (for translations and scales)
```

Comment:
I assume, either `outputsVec4` or `outputsVec3` are used, never both?
Shouldn't it be a `std::variant<std::vector<glm::vec4>, std::vector<glm::vec3>>`, or a `std::vector<std::variant<glm::vec4, glm::vec3>>` then?

Moreover, inputs and outputs are probably meant to have the same size?
Maybe something like `std::vector<KeyFrame> keyFrames;`, instead, with
```
struct KeyFrame
{
    float time;
    std::variant<glm::vec4, glm::vec3> value;
};
```